core/domain/topic_services.py

```diff
@@ -189,6 +193,185 @@ def save_new_topic(committer_id, topic):
         }])
 
 
+def apply_change_list(topic_id, change_list):
+    """Applies a changelist to a topic and returns the result.
+
+    Args:
+        topic_id: str. ID of the given topic.
+        change_list: list(dict). A change list to be applied to the given
+            topic. Each entry in change_list is a dict that represents a
+            TopicChange object.
+
+    Returns:
+      Topic. The resulting topic domain object.
+    """
+    topic = get_topic_by_id(topic_id)
+    try:
+        changes = [topic_domain.TopicChange(change_dict)
+                   for change_dict in change_list]
+
+        for change in changes:
+            if change.cmd == topic_domain.CMD_UPDATE_TOPIC_PROPERTY:
+                if (change.property_name ==
+                        topic_domain.TOPIC_PROPERTY_NAME):
+                    topic.update_name(change.new_value)
+                elif (change.property_name ==
+                      topic_domain.TOPIC_PROPERTY_DESCRIPTION):
+                    topic.update_description(change.new_value)
+                elif (change.property_name ==
+                      topic_domain.TOPIC_PROPERTY_CANONICAL_STORY_IDS):
+                    topic.update_canonical_story_ids(change.new_value)
+                elif (change.property_name ==
+                      topic_domain.TOPIC_PROPERTY_ADDITIONAL_STORY_IDS):
+                    topic.update_additional_story_ids(change.new_value)
+                elif (change.property_name ==
+                      topic_domain.TOPIC_PROPERTY_SKILL_IDS):
+                    topic.update_skill_ids(change.new_value)
+                elif (change.property_name ==
+                      topic_domain.TOPIC_PROPERTY_LANGUAGE_CODE):
+                    topic.update_language_code(change.new_value)
+        return topic
+
+    except Exception as e:
+        logging.error(
+            '%s %s %s %s' % (
+                e.__class__.__name__, e, topic_id, change_list)
+        )
+        raise
+
+
+def _save_topic(committer_id, topic, commit_message, change_list):
+    """Validates a topic and commits it to persistent storage. If
```

Comment:
Here and in the save() methods for the other objects: where does the validation actually happen?

Reply:
Done, added the validations.